bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -885,6 +958,27 @@ protected void setSelectionToWidget(@SuppressWarnings("rawtypes") List list, boo
 					System.arraycopy(indices, 0, indices = new int[count], 0, count);
 				}
 
+				// item to select may be hidden inside expandable node.
+				if (indices.length < list.size() && getLastElement() instanceof IExpandableNode expNode) {
+					Object[] remEles = expNode.getRemainingElements();
+					boolean breakOuter = false;
+					for (Object searchItem : list) {
```

Comment:
I believe we should iterate not entire `list` but only starting from last found index above.
I couldn't see breakpoint hit here, do you know in which use case in the IDE or test this code is used?

Reply:
fixed.
Now we iterate over only unfound elements in the list.
I too don't know where on IDE this logic works. But i have updated `Snippet068TableViewerWithLimit` to demo this.